src/ucp/proto/proto_am.inl

```diff
@@ -86,6 +122,13 @@ ucs_status_t ucp_do_am_bcopy_multi(uct_pending_req_t *self, uint8_t am_id_first,
             packed_len = uct_ep_am_bcopy(uct_ep, am_id_first, pack_first, req, 0);
             UCS_PROFILE_REQUEST_EVENT_CHECK_STATUS(req, "am_bcopy_first",
                                                    packed_len, packed_len);
+            if (ucs_unlikely(packed_len == UCS_ERR_NO_RESOURCE) &&
+                (am_flags & UCP_AM_SEND_FLAG_COPY_HEADER)) {
```

Comment:
Hi. Yes. I'm a bit confused... It was changed because you suggested it:
https://github.com/openucx/ucx/pull/8173#discussion_r890203356

At the end we need to check the am_flags param because "ucp_do_am_bcopy_multi" is common, this check will also save the call for copy_header in some cases if i'm not mistaken, maybe this is the reason you suggested it before?

Reply:
The reason of original suggestion is lost due to force push, but changes should be consistent for all usages.
IIRC I sugested to check it ouside because the function was called "copy_header"
and the code was expcted to be like
```
if (am_flags & UCP_AM_SEND_FLAG_COPY_HEADER) {
    copy_header(...);
}
```

Reply:
I see.
IMO this variable should be name is_am, it's purpose is to check if this request is AM, nothing more, and we need to pass it UCP_REQUEST_FLAG_SEND_AM when calling from AM flow.